engine/runtime/src/main/java/org/enso/interpreter/runtime/state/ContextPermissions.java

```diff
@@ -0,0 +1,4 @@
+package org.enso.interpreter.runtime.state;
+
+/** Fields correspond to the constructors of {@code Standard.Base.Runtime.Context} builtin type. */
+record ContextPermissions(boolean input, boolean output, boolean dataflowStacktrace) {}
```

Comment:
Such a simple class can certainly remain encapsulated in its only user - e.g. in `ExecutionEnvironment`.

Reply:
When I [asked for refactoring](https://github.com/enso-org/enso/pull/11153/files#r1803027966) I was talking about moving the code, not making the simple `record` its own `.java` file.

Reply:
It is also used by `WithContextNode`. So we need to keep it package private.